File: app/src/main/java/com/legue/axel/lolsummonertool/database/converter/DoubleMapConverters.java
package com.legue.axel.lolsummonertool.database.converter;

import android.arch.persistence.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.Collections;
import java.util.Map;

public class DoubleMapConverters {

    @TypeConverter
    public static Map<String, Double> fromString(String value) {
        Gson gson = new Gson();

        if (value == null) {
            return Collections.emptyMap();
        }

        Type mapType = new TypeToken<Map<String, Double>>() {
        }.getType();
        return gson.fromJson(value, mapType);
    }

    @TypeConverter
    public static String fromDoubleMap(Map<String, Double> map) {
        Gson gson = new Gson();
        return gson.toJson(map);
    }
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/converter/DoubleMapConverters.kt
package com.legue.axel.lolsummonertool.database.converter

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class DoubleMapConverters {

    @TypeConverter
    fun fromString(value: String?): Map<String, Double>? {

        if (value == null || value == "null") {
            return emptyMap()
        }

        val mapType = object : TypeToken<Map<String, Double>>() {}.type
        val objects = Gson().fromJson(value, mapType) as Map<String, Double>
        return objects.toMap()
    }

    @TypeConverter
    fun fromDoubleMap(map: Map<String, Double>): String {
        return Gson().toJson(map)
    }
}
